csharp_dotnetcore/8.Suggested-Actions/Suggested_Actions/SuggestedActionsBot.cs

```diff
@@ -0,0 +1,121 @@
+                                "Blue is the best color, I agree.",
+                                cancellationToken: cancellationToken);
+                            break;
+                        }
+
+                        default:
+                        {
+                            await turnContext.SendActivityAsync(
+                                $"Please select a color from the suggested actions",
+                                cancellationToken: cancellationToken);
+                            break;
+                        }
+                    }
+
+                    await SendSuggestedActionsAsync(turnContext, cancellationToken);
+                    break;
+                case ActivityTypes.ConversationUpdate:
+
+                    // Send a welcome message to the user and tell them what actions they need to perform to use this bot
+                    if (turnContext.Activity.MembersAdded.Any())
+                    {
+                        {
+                            foreach (var member in turnContext.Activity.MembersAdded)
+                            {
+                                var newUserName = member.Name;
+                                if (member.Id != turnContext.Activity.Recipient.Id)
+                                {
+                                    await turnContext.SendActivityAsync(
+                                        $"Welcome to SuggestedActionsBot {member.Name}." +
+                                        $" This bot will introduce you to suggestedActions." +
+                                        $"  Please answer the question:",
+                                        cancellationToken: cancellationToken);
+                                    await SendSuggestedActionsAsync(turnContext, cancellationToken);
+                                }
+                            }
+                        }
+                    }
+
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Creates and sends an activity with suggested actions to the user.
+        /// </summary>
+        /// <param name="turnContext">Provides the context for the turn of the bot.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A <see cref="Task"/> representing the operation result of the operation.</returns>
+        private static async Task SendSuggestedActionsAsync(ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            var reply = turnContext.Activity.CreateReply("What is your favorite color");
+            reply.SuggestedActions = new SuggestedActions()
+            {
+                Actions = new List<CardAction>()
+                {
+                    new CardAction() { Title = "Red", Type = ActionTypes.ImBack, Value = "Red" },
```

Comment:
Comments. For example, what is ImBack and why was it chosen?

Reply:
Added these comments to the summary section of this method.  It made more sense there as I could link ActionTypes via a 'see' reference